astroid/brain/brain_typing.py

```diff
@@ -316,6 +316,9 @@ def infer_typing_alias(
         # This is an issue in cases where the aliased class implements it,
         # but the typing alias isn't subscriptable. E.g., `typing.ByteString` for PY39+
         _forbid_class_getitem_access(class_def)
+
+    # Avoid re-instantiating this class every time it's seen
+    node._explicit_inference = lambda node, context: iter([class_def])
```

Comment:
This feels like something we might need to do for a lot of brains.

We us a similar pattern for the `enum` brain. :sweat:

Reply:
You're quite right. I traced the bulk of the primer noise to needing to do this in other places in the typing brain.